### PR TITLE
Tracking: Add hardening for user role retrieval

### LINKED ISSUE #11352: Tracking: Add hardening for user role retrieval
## Feature Description

According to [this report](https://wordpress.org/support/topic/php-warning-tracking-php-175/), there can be a PHP warning triggered here:

https://github.com/GoogleForCreators/web-stories-wp/blob/cefc2736398ab391e3fbd0a437b675b83bfa3039/includes/Tracking.php#L175

```
PHP Warning: Undefined array key 0 in /var/web/site/public_html/wp-content/plugins/web-stories/includes/Tracking.php on line 175
```

It's not clear to me how that can happen, but I suppose in theory a user can have zero roles, or the array can have non-numeric indices.

For extra hardening we can

* Wrap `wp_get_current_user()->roles` in `array_values()`
* Use an `isset()` check when trying to retrieve the first role

### PR DESCRIPTION
**Pull request by timarney**
## Context

According to [this report](https://wordpress.org/support/topic/php-warning-tracking-php-175/), there can be a PHP warning triggered here:

https://github.com/GoogleForCreators/web-stories-wp/blob/cefc2736398ab391e3fbd0a437b675b83bfa3039/includes/Tracking.php#L175

```
PHP Warning: Undefined array key 0 in /var/web/site/public_html/wp-content/plugins/web-stories/includes/Tracking.php on line 175
```

## Summary

Updates check to not depend on specific index 

## Testing Instructions

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

Created a code snippet to test this ...

```php
function wp_get_current_user(){
    $user = new stdClass();
    $user->roles = array("admin");
    return $user;
}

// PHP Notice:  Undefined offset: 1
// $role        = ! empty( wp_get_current_user()->roles ) ? wp_get_current_user()->roles[1] : ''; 

// works as expected
$role        = ! empty( wp_get_current_user()->roles ) && \is_array( wp_get_current_user()->roles ) ? array_shift( wp_get_current_user()->roles ) : '';  


echo '--role: '.$role.' --';
```

### Does this PR have a security-related impact?

No 

### Does this PR change what data or activity we track or use?

No 

### Does this PR have a legal-related impact?

No 

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #11352

### CODE REVIEW

File: includes/Tracking.php [hunk 1]
```diff
@@ -172,7 +172,7 @@ public function get_settings(): array {
 	 * @return array User properties.
 	 */
 	private function get_user_properties(): array {
-		$role        = ! empty( wp_get_current_user()->roles ) ? wp_get_current_user()->roles[0] : '';
+		$role        = ! empty( wp_get_current_user()->roles ) && \array_key_exists( 0, wp_get_current_user()->roles ) ? wp_get_current_user()->roles[0] : '';
```
Comment (spacedmonkey) [inline includes/Tracking.php]:
```suggestion
		$role        = ! empty( wp_get_current_user()->roles ) && is_array( wp_get_current_user()->roles ) ? array_shift( wp_get_current_user()->roles ) : ''; 
```

I prefer to use an array_shift here, over using the first element in an array. This is a little cleaner as well.

**Review by spacedmonkey — DISMISSED** [review]
Lint error.

**Comment by timarney** [comment]
>Lint error

Fixed

**Comment by swissspidy** [comment]
Possible to add a test for this? Otherwise looks good.

**Comment by timarney** [comment]
> Possible to add a test for this? Otherwise looks good.

Not sure how to mock out the an object with missing index (0) properly 

using the original code  non of these throw 

```php 
$user->roles = array(); 
$user->roles = false; 
$user->roles = ""; 
```

https://developer.wordpress.org/reference/classes/wp_user/

**Comment by swissspidy** [comment]
This should work:

```php
/**
 * @covers ::get_user_properties
 */
public function test_get_settings_user_malformed_roles(): void {
	$user_id = self::factory()->user->create(
		[
			'role' => 'administrator',
		]
	);

	wp_set_current_user( $user_id );

	global $current_user;
	$current_user->roles = [ 1 => 'administrator' ];

	$this->site_kit->method( 'get_plugin_status' )->willReturn(
		[
			'installed'       => true,
			'active'          => true,
			'analyticsActive' => true,
			'link'            => 'https://example.com',
		]
	);
	$this->experiments->method( 'get_enabled_experiments' )
		->willReturn( [ 'enableFoo', 'enableBar' ] );

	$settings = $this->instance->get_settings();

	$this->assertSame( 'administrator', $settings['userProperties']['userRole'] );
}
```

**Review by swissspidy — APPROVED** [review]

**Pull request merged**